packages/patternfly-4/react-core/src/demos/BulkSelectTable.md

```diff
@@ -0,0 +1,225 @@
+---
+title: 'Bulk select table'
+section: 'demos'
+---
+
+## Bulk select table demo
+
+import { Pagination, PaginationVariant, Title, Checkbox } from '@patternfly/react-core';
+import { Table, TableHeader, TableBody} from '@patternfly/react-table';
+
+```js
+import React from 'react';
+import { Pagination, PaginationVariant, Title, Checkbox } from '@patternfly/react-core';
+import { Table, TableHeader, TableBody } from '@patternfly/react-table';
+
+class BulkSelectTableDemo extends React.Component {
+  constructor(props) {
+    this.state = {
+      res: [],
+      perPage: 20,
+      total: 100,
+      page: 1,
+      error: null,
+      loading: true,
+      selectedItems: [],
+      numSelected: 0,
+      isDropDownOpen: false,
+      isKebabOpen: false,
+      searchValue: ''
+    };
+
+    this.onSelect = (event, isSelected, rowId) => {
+      const { selectedItems } = this.state;
+      const rows = [...this.state.res];
+      const id = rows[rowId].id;
+      rows[rowId].selected = isSelected;
+      this.setState({
+        res: rows,
+        selectedItems: isSelected ? [...selectedItems, id] : selectedItems.filter(itemId => itemId !== id)
+      });
+    };
+
+    this.updateSelected = () => {
+      const { res, selectedItems } = this.state;
+      let rows = res.map(post => {
+        post.selected = selectedItems.includes(post.id);
+        return post;
+      });
+
+      this.setState({
+        res: rows
+      });
+    };
+
+    this.handleSelectClick = newState => {
+      if (newState === 'none') {
+        this.setState(
+          {
+            selectedItems: []
+          },
+          this.updateSelected
+        );
+      } else if (newState === 'page') {
+        let newRows = [];
+        let rows = this.state.res.map(post => {
+          const isSelected = post.selected;
+          newRows = isSelected ? [...newRows] : [...newRows, post.id];
+          post.selected = true;
+          return post;
+        });
+        this.setState(
+          {
+            selectedItems: this.state.selectedItems.concat(newRows)
+          },
+          this.updateSelected
+        );
+      } else {
+        const { total } = this.state;
+        let newRows = [];
+        for (var i = 1; i <= total; i++) newRows = [...newRows, i];
+
+        this.setState(
+          {
+            selectedItems: newRows
+          },
+          this.updateSelected
+        );
+      }
+    };
+
+    this.onDropDownToggle = isOpen => {
+      this.setState({
+        isDropDownOpen: isOpen
+      });
+    };
+
+    this.onDropDownSelect = event => {
+      this.setState({
+        isDropDownOpen: !this.state.isDropDownOpen
```

Comment:
Can we use the [function form of setState](https://medium.com/@wisecobbler/using-a-function-in-setstate-instead-of-an-object-1f5cfd6e55d1) instead? Might help prevent some bugs down the road.